packages/calling/src/CallingClient/line/line.test.ts

```diff
@@ -0,0 +1,203 @@
+  const backupMobiusUris = jest.fn(() => mobiusUris.backup);
+  const userId = webex.internal.device.userId;
+  const clientDeviceUri = webex.internal.device.url;
+
+  const handleErrorSpy = jest.spyOn(utils, 'handleRegistrationErrors');
+
+  jest.clearAllMocks();
+
+  describe('Registration tests', () => {
+    let line;
+    const mockRegistrationBody = getMockDeviceInfo();
+
+    const discoveryBody: MobiusServers = getMobiusDiscoveryResponse();
+    const primaryUrl = `${discoveryBody.primary.uris[0]}/calling/web/`;
+
+    const registrationPayload = <WebexRequestPayload>(<unknown>{
+      statusCode: 200,
+      body: mockRegistrationBody,
+    });
+
+    const mockBody = {
+      userId: webex.internal.device.userId,
+      clientDeviceUri: webex.internal.device.url,
+      serviceData: {
+        domain: '',
+        indicator: 'calling',
+      },
+    };
+
+    beforeEach(() => {
+      line = new Line(
+        userId,
+        clientDeviceUri,
+        LineStatus.ACTIVE,
+        mutex,
+        primaryMobiusUris(),
+        backupMobiusUris(),
+        LOGGER.INFO
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllTimers();
+      jest.clearAllMocks();
+      jest.useRealTimers();
+    });
+
+    it('verify successful Registration cases and keepalive', async () => {
+      jest.useFakeTimers();
+      webex.request.mockReturnValue(registrationPayload);
+
+      expect(createRegistrationSpy).toBeCalledOnceWith(
+        webex,
+        defaultServiceData,
+        expect.any(Mutex),
+        expect.anything(),
+        LOGGER.INFO
+      );
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.DEFAULT);
+      await line.register();
+
+      expect(webex.request).toBeCalledOnceWith({
+        ...getMockRequestTemplate(),
+        body: {
+          userId,
+          clientDeviceUri,
+          serviceData: defaultServiceData,
+        },
+        uri: `${primaryUrl}device`,
+        method: 'POST',
+      });
+      expect(handleErrorSpy).not.toBeCalled();
+
+      expect(line.getRegistrationStatus()).toEqual(MobiusStatus.ACTIVE);
+      expect(line.getActiveMobiusUrl()).toEqual(primaryUrl);
+      expect(line.getLoggingLevel()).toEqual(LOGGER.INFO);
+      expect(line.getDeviceId()).toEqual(mockRegistrationBody.device.deviceId);
+
+      /** trigger sendKeepAlive and verify whether keepalive request is sent or not */
+      webex.request.mockClear();
+
+      /* specify keepalive interval as 30 seconds and advance timers by 30 */
+      line.sendKeepAlive({...mockRegistrationBody, keepaliveInterval: 30});
+
+      jest.advanceTimersByTime(30 * 1000);
+      await Promise.resolve();
+
+      expect(webex.request).toBeCalledOnceWith({
+        ...getMockRequestTemplate(),
+        uri: `${mockRegistrationBody.device.uri}/status`,
+        method: 'POST',
+      });
+      jest.useRealTimers();
+    });
+
+    it('verify post registration api call when default mobius url is used', async () => {
```

Comment:
I feel it does not matter to line class which url is being used.
Ideally CallingClient would take care of deciding the primaryUrl and backupUrl's and adding a default url incase of discovery failure. This should have been a test case in CallingClient. Even registration class would be unaware of presence of a default Url. So this is purely a CallingClient's test case.

Reply:
Yeah removed it from line test and default mobius url block is already covered in existing calling client tests.